eZ/Publish/API/Repository/ObjectStateService.php

```diff
@@ -47,6 +47,31 @@ public function createObjectStateGroup(ObjectStateGroupCreateStruct $objectState
      */
     public function loadObjectStateGroup($objectStateGroupId, array $prioritizedLanguages = []);
 
+    /**
+     * Loads an object state by identifier and group it belongs to.
+     *
+     * @param string $identifier
+     * @param mixed $groupId
+     * @param string[] $prioritizedLanguages Used as prioritized language code on translated properties of returned object.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the state was not found
+     *
+     * @return \eZ\Publish\API\Repository\Values\ObjectState\ObjectState
+     */
+    public function loadByIdentifier($identifier, $groupId, array $prioritizedLanguages = []);
```

Comment:
```suggestion
    public function loadObjectStateByIdentifier(string $groupIdentifier, string $stateIdentifier, array $prioritizedLanguages = []);
```

or 

```suggestion
    public function loadObjectStateByIdentifier(ObjectStateGroup $group, string $identifier, array $prioritizedLanguages = []);
```

WDYT @alongosz  ?

Reply:
Same as above.  I'm fine either way. I just followed the example in the existing spi handler.  If we change them here, should we change it their too?

Reply:
@wizhippo well, if its used already it would mean we need to expose additional spi methods afaik.

Reply:
Side note: also add changes to https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/REST/Client/ObjectStateService.php#L289

Reply:
Yes, API should not operate on raw IDs when there's a possibility to use unique identifiers.